vgaidarji/react-graphql-playground
# Add Cypress E2E tests to the project

### What has been done
1. Add [Cypress](https://www.cypress.io/) E2E tests to the project. Covered full E2E flow of navigation for critical path (Home -> Countries List -> Country page).
2. Included Cypress tests into GH actions workflow

### How to test
1. Run `npx cypress open` and confirm E2E suite (`cypress/e2e/full-e2e-spec.cy.ts` spec) passes successfully.
2. Confirm Cypress tests are passing successfully on GH Actions as well and artifacts are being archived. 
	**e2e tests part of workflow**
	- <img width="1053" alt="image" src="https://user-images.githubusercontent.com/3036347/230072276-5d0214d5-73eb-4f4c-877b-64d8ac69af1c.png">
	**Archived e2e test recording**
	- <img width="1362" alt="image" src="https://user-images.githubusercontent.com/3036347/230072576-68c50ede-76d8-4cd9-b555-9997ca93aa86.png">
	- <video src="https://user-images.githubusercontent.com/3036347/230072611-8e42c9a8-6f1b-47fa-884c-af2b5d22fccb.mp4" controls="controls" style="max-width: 730px;">
</video>
